aws_lambda_builders/workflows/rust_cargo/DESIGN.md

```diff
@@ -0,0 +1,44 @@
+# Rust Cargo Builder
+
+## Scope
+
+This package enables the creation of a Lambda deployment package for Rust projects managed using the [cargo](https://doc.rust-lang.org/cargo/) build tool targeting Lambda's "provided" runtime. Rust support for the provided runtime is bundled as a compilation dependency of these projects, provided by the [lambda](https://github.com/awslabs/aws-lambda-rust-runtime) crate.
+
+## Implementation
+
+The general algorithm for preparing a rust executable for use on AWS Lambda
+is as follows.
+
+### Build
+
+It builds a binary in the standard cargo target directory. 
+
+### Copy and Rename executable
+
+It then copies the executable to the target directory renaming the executable to "bootstrap" to honor the provided runtime's [expectation on executable names](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html).
+
+## Challenges
+
+### Cross compilation
+
+Cargo builds binaries targeting host platforms. When the host platform is not the same as the target platform it leverages a parameterized notion of a named target, typically installed and managed by [rustup](https://github.com/rust-lang/rustup). In the case of `sam` we default this target to to `x86_64-unknown-linux-musl` when the host is not linux based.
+
+Users can simply run `rustup target add x86_64-unknown-linux-musl` if needed. It's also possible for sam to detect this when needed and do that for users by parsing the output of `rustup target list --installed` but it would be unusual for sam to
```

Comment:
I reckon that with the new provided.al2, all references to MUSL should be removed for simplicity since [`x86_64-unknown-linux` works fine, potentially faster and with less object size issues](https://github.com/awslabs/aws-lambda-rust-runtime/discussions/306#discussioncomment-485478)?